### PR TITLE
feat(frontend/shared): change password funcionality in profile screen bb-417

### PR DESCRIPTION
**Pull request by iqbalalayubbi**
**Issue Fixed**
New password should match with confirm new password

**Demo**

https://github.com/user-attachments/assets/8ec66f4b-95cd-4a75-912a-de6acea3cf85

### CODE REVIEW

File: apps/frontend/src/pages/profile/libs/components/update-password-form/update-password-form.tsx [hunk 1]
```diff
@@ -21,17 +22,29 @@ const UpdatePasswordForm: React.FC<Properties> = ({ onSubmit }: Properties) => {
...
 		(event_: React.BaseSyntheticEvent): void => {
-			void handleSubmit(onSubmit)(event_);
+			void handleSubmit((payload: UserUpdatePasswordRequestDto) => {
+				const { confirmNewPassword } = getValues();
+				const { newPassword } = payload;
+
+				if (confirmNewPassword === newPassword) {
+					onSubmit(payload);
+				} else {
+					setError("confirmNewPassword", {
```
Comment (fshabanov) [inline apps/frontend/src/pages/profile/libs/components/update-password-form/update-password-form.tsx:40]: Move the name to enum as well

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: packages/shared/src/modules/auth/libs/enums/confirm-password-custom-validation.enum.ts [hunk 2]
```diff
@@ -1,7 +1,10 @@
 const ConfirmPasswordCustomValidation = {
 	ERROR_MESSAGE: "Passwords do not match. Please re-enter your password",
 	ERROR_TYPE: "custom",
-	FIELD: "confirmPassword",
+	FIELDS: {
+		confirmNewPassword: "confirmNewPassword",
+		confirmPassword: "confirmPassword",
+	},
```
Comment (fshabanov) [inline packages/shared/src/modules/auth/libs/enums/confirm-password-custom-validation.enum.ts:7]:
```suggestion
	FIELD: {
		CONFIRM_NEW_PASSWORD: "confirmNewPassword",
		CONFIRM_PASSWORD: "confirmPassword",
	},
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/mobile/src/screens/auth/components/sign-up-form/sign-up-form.tsx [hunk 3]
```diff
@@ -48,12 +48,14 @@ const SignUpForm: React.FC<Properties> = ({ onSubmit }) => {
 	const handleFormSubmit = useCallback((): void => {
 		void handleSubmit((signUpSubmissionData: UserSignUpFormDto) => {
 			const { password } = signUpSubmissionData;
-			const confirmPassword = watch(ConfirmPasswordCustomValidation.FIELD);
+			const confirmPassword = watch(
```
Comment (fshabanov) [inline apps/mobile/src/screens/auth/components/sign-up-form/sign-up-form.tsx:51]: Update this to use `getValues`, instead of `watch`

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/mobile/src/screens/auth/components/sign-up-form/sign-up-form.tsx (hunk 3 above)
Comment (iqbalalayubbi) [reply apps/mobile/src/screens/auth/components/sign-up-form/sign-up-form.tsx:51]: image

Is there something wrong?

Comment (fshabanov) [reply apps/mobile/src/screens/auth/components/sign-up-form/sign-up-form.tsx:51]: Yes, check `update-password-form.tsx`

**Review by fshabanov — APPROVED** [review]

**Pull request merged**